File: Python/scraping/extract_links.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("https://example.com")
    links = page.eval_on_selector_all("a", "els => els.map(el => el.href)")
    print(links)
    browser.close()

File: NodeJS/scraping/extract_links.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('https://example.com');
  const links = await page.$$eval('a', els => els.map(el => el.href));
  console.log(links);
  await browser.close();
})();
